### ISSUE TITLE
Support maps > 64x64

### ISSUE THREAD

**Issue by stefanhendriks** [stefanhendriks/Dune-II---The-Maker#62]
Ofcourse, keeping support for the official 64x64 (or rather 62x62) maps... but for skirmish we could go up to way higher.

The dimensions could be fixed, or flexible. Perhaps up to 512x512, or even 1024x1024?

Would need to think about how to go about the minimap.

- [x] concentrate functions and data in one class (ie get rid of `cCellCalculator`, merge it within `Map` class)
- [x] get rid of the generic `iCellMake` and such functions (in `common.*`)
- [x] bullets should use absolute position coordinates & being rendered with them
- [x] structures should use absolute position coordinates & being rendered with them
- [x] units should use absolute position coordinates & being rendered with them
- [x] particles should use absolute position coordinates & being rendered with them
- [x] render maps < 64x64 maps properly on minimap (can be further improved, see #162)
- [x] test campaign missions
- [x] test skirmish missions (respect their dimensions - see Skirmish map: "Cliffs of Rene")
- [x] make it explicit what the map playable borders are (basically dimension - 2) and what the data dimensions are (ie, the 64x64 vs 62x62 case)
- [x] generate random maps of 128x128 instead of 64x64 (later configurable after #106 )
- [x] bigger maps than 128x128? (deferred to #161)

**Comment by stefanhendriks** [comment]
Perhaps we can store data more efficiently, see also https://gameprogrammingpatterns.com/flyweight.html
